lib/internal/test_runner/test.js

```diff
@@ -523,6 +523,7 @@ class Test extends AsyncResource {
 
     // The test has run, so recursively cancel any outstanding subtests and
     // mark this test as failed if any subtests failed.
+    this.pendingSubtests = [];
```

Comment:
this should probably use defineProperty?

Reply:
The object is never exposed to user land.
See how it it originally defined in the constructor

Reply:
AsyncResource instances inherit from Object.prototype, so by adding a setter to Object.prototype for “pendingSubtests” would allow someone to intercept this part of the class.

Reply:
It's overriding a property that was already defined in the constructor, so the constructor itself should be using `defineProperty` (or the `setOwnProperty` helper from https://github.com/nodejs/node/pull/44007), this line is fine though.

Reply:
il do this in a follow-up pr

Reply:
An alternative would be to use a class field in AsyncResource (and/or here)